hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/util/TestStringUtils.java

```diff
@@ -566,6 +568,86 @@ public void testStringCollectionSplitByEquals() {
         .containsEntry("element.xyz.key5", "element.abc.val5")
         .containsEntry("element.xyz.key6", "element.abc.val6")
         .containsEntry("element.xyz.key7", "element.abc.val7");
+
+    try {
+      StringUtils.getTrimmedStringCollectionSplitByEquals(
+          " = element.abc.val1");
+      throw new RuntimeException("Expected to throw IllegalArgumentException");
+    } catch (IllegalArgumentException e) {
+      Assertions
+          .assertThat(e)
+          .describedAs("Exception thrown due to illegal arguments")
+          .hasMessageStartingWith(STRING_COLLECTION_SPLIT_EQUALS_INVALID_ARG);
+    }
+
+    splitMap = StringUtils.getTrimmedStringCollectionSplitByEquals(
+        "element.first.key1 = element.first.val2 ,element.first.key1 =element.first.val1");
+    Assertions
+        .assertThat(splitMap)
+        .describedAs("Map of key value pairs split by equals(=) and comma(,)")
+        .hasSize(1)
+        .containsEntry("element.first.key1", "element.first.val1");
+
+    splitMap = StringUtils.getTrimmedStringCollectionSplitByEquals(
+        ",,, , ,, ,element.first.key1 = element.first.val2 ,"
+            + "element.first.key1 = element.first.val1 , ,,, ,");
+    Assertions
+        .assertThat(splitMap)
+        .describedAs("Map of key value pairs split by equals(=) and comma(,)")
+        .hasSize(1)
+        .containsEntry("element.first.key1", "element.first.val1");
+
+    try {
+      StringUtils.getTrimmedStringCollectionSplitByEquals(
```

Comment:
LambdaTestUtils.intercept(). It's predictable I will insist on that, so save time in future. and it includes a string value check. so all you need is 

```java
intercept(IllegalArgumentException.class,
  STRING_COLLECTION_SPLIT_EQUALS_INVALID_ARG, () ->
    StringUtils.getTrimmedStringCollectionSplitByEquals(
          "element.abc.key1="));
```

See: much better. and if the closure doesn't fail, will even include the string value of the result in the exception raised.

Reply:
ah yes, that's already an amazing utility